### PR TITLE
Validation improvement

### PR DESCRIPTION
**Pull request by Amit-Singh40**
**Change log description**  
PSC code lacks general parameter and argument validation.  Using Preconditions to validate arguments

**Purpose of the change**  
Validation improvement. 

**What the code does**  
Improves validation and parameter checking

**How to verify it**  
NA

### CODE REVIEW

**Review by sachin-j-joshi — APPROVED** [review]
LGTM

**Review by sachin-j-joshi — APPROVED** [review]

File: pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/DeviceDriver.java [hunk 1]
```diff
@@ -30,7 +31,7 @@ public abstract class DeviceDriver extends AbstractService implements AutoClosea
     private static final String CREATE_SCOPE_KEY = "CREATE_SCOPE";
 
     public DeviceDriver(DeviceDriverConfig config) {
-        this.config = config;
+        this.config = Preconditions.checkNotNull(config, "deviceDriverConfig");
```
Comment (abhinb) [inline pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/DeviceDriver.java:34]: What message would it print when null? with "deviceDriverConfig" as second param?

**Review by abhinb — APPROVED** [review]
maybe we can also look at any asserts in the logic if any that we can add.

File: pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/DeviceDriver.java (hunk 1 above)
Comment (Amit-Singh40) [reply pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/DeviceDriver.java:34]: Below is the output
```
Exception in thread "main" java.lang.NullPointerException: deviceDriverConfig
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:907)
	at io.pravega.sensor.collector.DeviceDriver.<init>(DeviceDriver.java:34)
	at io.pravega.sensor.collector.simple.SimpleDeviceDriver.<init>(SimpleDeviceDriver.java:56)
	at io.pravega.sensor.collector.accelerometer.AccelerometerDriver.<init>(AccelerometerDriver.java:44)
```

**Comment by Amit-Singh40** [comment]
> maybe we can also look at any asserts in the logic if any that we can add.

sure

**Pull request merged**